packages/table/src/quadrants/tableQuadrantStack.tsx

```diff
@@ -190,46 +228,58 @@ export class TableQuadrantStack extends AbstractComponent<ITableQuadrantStackPro
         //   this keeps every sync'd quadrant visually aligned in each animation frame.
         this.throttledHandleMainQuadrantScroll = CoreUtils.throttleReactEventCallback(this.handleMainQuadrantScroll);
         this.throttledHandleWheel = CoreUtils.throttleReactEventCallback(this.handleWheel, { preventDefault: true });
+
+        this.cache = new TableQuadrantStackCache();
     }
 
     /**
      * Scroll the main quadrant to the specified scroll offset, keeping all other quadrants in sync.
      */
     public scrollToPosition(scrollLeft: number, scrollTop: number) {
         const { scrollContainer } = this.quadrantRefs[QuadrantType.MAIN];
+
         this.wasMainQuadrantScrollChangedFromOtherOnWheelCallback = false;
+
         // this will trigger the main quadrant's scroll callback below
         scrollContainer.scrollLeft = scrollLeft;
         scrollContainer.scrollTop = scrollTop;
+
+        this.syncQuadrantViews();
     }
 
     public componentDidMount() {
         this.emitRefs();
-        this.syncQuadrantSizes();
-        this.syncQuadrantMenuElementWidths();
-        CoreUtils.safeInvoke(this.props.columnHeaderRef, this.findColumnHeader(QuadrantType.MAIN));
-        CoreUtils.safeInvoke(this.props.rowHeaderRef, this.findRowHeader(QuadrantType.MAIN));
+        this.syncQuadrantViews();
     }
 
-    public componentDidUpdate() {
-        this.emitRefs();
-        this.syncQuadrantSizes();
-        this.syncQuadrantMenuElementWidths();
-        CoreUtils.safeInvoke(this.props.columnHeaderRef, this.findColumnHeader(QuadrantType.MAIN));
-        CoreUtils.safeInvoke(this.props.rowHeaderRef, this.findRowHeader(QuadrantType.MAIN));
+    public componentDidUpdate(prevProps: ITableQuadrantStackProps) {
+        // sync'ing quadrant views triggers expensive reflows, so we only call
+        // it when layout-affecting props change.
+        if (
+            this.props.numFrozenColumns !== prevProps.numFrozenColumns ||
+            this.props.numFrozenRows !== prevProps.numFrozenRows ||
```

Comment:
style: I prefer putting the boolean operator at the _beginning_ of the line so it's immediately recognizable as a continuation of the previous line.

```js
if (
    A
    || B
    || C
    || D
) {
```

Reply:
Me too, but `prettier` doesn't. :(